### PR TITLE
fix tck compare in order

### PR DESCRIPTION
**Pull request by HarrisChu**
1. fix compare in order, using _compate_order_list
2. fix typo  
3. change `the first {n:d} records in order` with Equal type.

### CODE REVIEW

**Review by Aiee — DISMISSED** [review]

**Review by czpmango — APPROVED** [review]
Nice fix!

**Review by yixinglu — DISMISSED** [review]

**Review by yixinglu — DISMISSED** [review]

**Review by CPWstatic — DISMISSED** [review]

**Review by czpmango — APPROVED** [review]

File: tests/tck/features/match/MultiLineMultiQueryParts.feature [hunk 1]
```diff
@@ -172,15 +172,6 @@ Feature: Multi Line Multi Query Parts
       | "Tim Duncan" | "Manu Ginobili"     | NULL |
       | "Tim Duncan" | "Manu Ginobili"     | NULL |
       | "Tim Duncan" | "Manu Ginobili"     | NULL |
-    When executing query:
```
Comment (yixinglu) [inline tests/tck/features/match/MultiLineMultiQueryParts.feature:175]: why remove this test case?

File: tests/tck/features/match/MultiLineMultiQueryParts.feature [hunk 2]
```diff
@@ -254,6 +245,7 @@ Feature: Multi Line Multi Query Parts
       | count |
       | 4     |
 
+  @skip
```
Comment (yixinglu) [inline tests/tck/features/match/MultiLineMultiQueryParts.feature:248]: And why to skip this scenario?

Comment (HarrisChu) [reply tests/tck/features/match/MultiLineMultiQueryParts.feature:248]: https://github.com/vesoft-inc/nebula/issues/4168

there's a bug, and the original test cases cannot run pass

**Review by yixinglu — APPROVED** [review]

**Review by Aiee — DISMISSED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by czpmango — APPROVED** [review]

**Pull request merged**